src/containers/BooksForm.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions';
+import CATEGORIES from '../utils/categories';
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: CATEGORIES[0],
+    };
+  }
+
+  handleCategoryChange(event) {
+    this.setState({
+      category: event.target.value,
+    });
+  }
+
+  handleTitleChange(event) {
+    this.setState({
+      title: event.target.value,
+    });
+  }
+
+  handleSubmit(event) {
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+    event.preventDefault();
+
+    const book = {
+      id: Math.random(),
```

Comment:
After all the debugging on slack, for some reason, you went offline.
I trust you can update this code before merging 👍 
 `id: Math.round(Math.random() * 100),`
Doing this you don't have to get float number as ID.